File: server/types/question.go
package types

import "time"

// Question ...
type Question struct {
	ID        int       `json:"id"`
	Text      string    `json:"text"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
}

// Questions ...
type Questions struct {
	Questions []Question `json:"questions"`
	TotalPage int        `json:"totalPage"`
}

File: front/src/types/question.ts
export interface IQuestion {
  id?: number;
  text: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface ISetQuestion {
  text: string;
}

export interface IGetQuestion {
  id: string;
}

export interface IGetQuestions {
  page?: number;
}

export interface IQuestions {
  questions?: IQuestion[];
  totalPage?: number;
  error?: string;
  loading: boolean;
}

export interface IOneQuestion {
  question?: IQuestion;
  error?: string;
  loading: boolean;
}
